code/modules/mod/mod_construction.dm

```diff
@@ -0,0 +1,164 @@
+/obj/item/mod/construction
+	desc = "A part used in MOD construction."
+
+/obj/item/mod/construction/helmet
+	name = "MOD helmet"
+	icon_state = "helmet"
+
+/obj/item/mod/construction/chestplate
+	name = "MOD chestplate"
+	icon_state = "chestplate"
+
+/obj/item/mod/construction/gauntlets
+	name = "MOD gauntlets"
+	icon_state = "gauntlets"
+
+/obj/item/mod/construction/boots
+	name = "MOD boots"
+	icon_state = "boots"
+
+/obj/item/mod/construction/shell
+	name = "MOD shell"
+	icon_state = "mod-construction"
+	desc = "An empty MOD shell."
+	var/build_step = 1
+
+/obj/item/mod/construction/core
+	name = "MOD core"
+	icon_state = "mod-core"
+	desc = "A mystical crystal able to convert cell power into energy usable by MODsuits."
+
+/obj/item/mod/construction/armor
+	name = "MOD standard armor plates"
+	desc = "Armor plates used to finish a MOD"
+	icon_state = "armor"
+	var/theme = /datum/mod_theme
+
+/obj/item/mod/construction/armor/engineering
+	name = "MOD engineering armor plates"
+	icon_state = "engineering-armor"
+	theme = /datum/mod_theme/engineering
+
+/obj/item/mod/paint
+	name = "MOD paint kit"
+	desc = "This kit will repaint your MODsuit to something unique."
+	icon = 'icons/obj/mod.dmi'
+	icon_state = "paintkit"
+
+/obj/item/mod/construction/shell/attackby(obj/item/I, mob/user, params)
+	. = ..()
+	var/atom/Tsec = drop_location()
+	switch(build_step)
+		if(1)
+			if(istype(I, /obj/item/mod/construction/core)) //Construct
+				if(!user.temporarilyRemoveItemFromInventory(I))
+					return
+				to_chat(user, "<span class='notice'>You insert [I] into [src].</span>")
+				build_step++
+				qdel(I)
```

Comment:
You should be preserving the core and such as you're building it, not deleting and making new ones.

Reply:
Here and everywhere else.

Reply:
Then you can check the conditions directly rather than incrementing a build step.